pkg/kubelet/kubelet.go

```diff
@@ -814,6 +818,8 @@ func (kl *Kubelet) syncPod(pod *api.BoundPod, dockerContainers dockertools.Docke
 		if err != nil {
 			glog.Errorf("Couldn't make a ref to pod %v, container %v: '%v'", pod.Name, container.Name, err)
 		}
+		kl.pullLock.RLock()
+		defer kl.pullLock.RUnlock()
```

Comment:
this won't work right.  the defer is scoped to the function.  The right answer is to make this use the pullImage function.